pkg/rtc/signalhandler.go

```diff
@@ -94,17 +94,35 @@ func HandleParticipantSignal(room types.Room, participant types.LocalParticipant
 	case *livekit.SignalRequest_UpdateMetadata:
 		var errorResponse *livekit.ErrorResponse
 		if participant.ClaimGrants().Video.GetCanUpdateOwnMetadata() {
-			err := room.UpdateParticipantMetadata(
-				participant,
+			if err := participant.CheckMetadataLimits(
 				msg.UpdateMetadata.Name,
 				msg.UpdateMetadata.Metadata,
 				msg.UpdateMetadata.Attributes,
-			)
-			if err != nil {
+			); err == nil {
+				if msg.UpdateMetadata.Name != "" {
```

Comment:
Just noticed that we are not allowing clearing of a set name/metadata. Is that by design?

Reply:
yeah. this is because with protobuf, we do not know if a string field is omitted vs intentionally set to ""

Reply:
name may make sense but not being able to clear metadata seems like a bug. i'm not sure how to fix this in a backward compatible way though... the field would need to be optional

Reply:
yeah, that is going to be tricky. Will get this merged and think about that.